### PR TITLE
[WIP] DHIS2-2382

### PR DESCRIPTION
**Pull request by stian-sandvold**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ObjectStyle.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.hisp.dhis.common;
+
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ObjectStyle.java]: License and author

File: dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/dialect/DhisPostgres.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * @author Morten Olav Hansen <mortenoh@gmail.com>
+ * @author Stian Sandvold <stian@dhis2.org>
+ */
+public class DhisPostgres
+    extends PostgreSQL95Dialect
+{
+    // ~ Constructors
+    // ========================================================================================================
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/dialect/DhisPostgres.java]: Remove ====

File: dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/dialect/DhisPostgres.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import org.hibernate.dialect.PostgreSQL95Dialect;
+
+import java.sql.Types;
+
+/**
+ * @author Morten Olav Hansen <mortenoh@gmail.com>
+ * @author Stian Sandvold <stian@dhis2.org>
+ */
+public class DhisPostgres
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/dialect/DhisPostgres.java]: Should this class be called DhisPostgresDialect ?

File: dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/jsonb/type/JsonBinaryType.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+import java.util.Properties;
+
+/**
+ * @author Morten Olav Hansen <mortenoh@gmail.com>
+ * @author Stian Sandvold <stian@dhis2.org>
+ */
+public class JsonBinaryType implements UserType, ParameterizedType
+{
+    // ~ Static instance fields
+    // ================================================================================================
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/jsonb/type/JsonBinaryType.java]: Remove === comments (these are clearly copied;)

File: dhis-2/dhis-support/dhis-support-jdbc/src/main/java/org/hisp/dhis/jdbc/dialect/StatementDialectFactoryBean.java [hunk 5]
```diff
@@ -51,6 +51,7 @@
         dialectMap = new HashMap<>();
         dialectMap.put( "org.hibernate.dialect.MySQLDialect", StatementDialect.MYSQL );
         dialectMap.put( "org.hibernate.dialect.PostgreSQLDialect", StatementDialect.POSTGRESQL );
+        dialectMap.put( "org.hisp.dhis.hibernate.dialect.DhisPostgres", StatementDialect.POSTGRESQL );
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-jdbc/src/main/java/org/hisp/dhis/jdbc/dialect/StatementDialectFactoryBean.java]: DhisPosgreSQLDialect instead?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataset/DataSet.java [hunk 6]
```diff
@@ -36,13 +36,7 @@
...
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Sets;
-import org.hisp.dhis.common.BaseDimensionalItemObject;
-import org.hisp.dhis.common.BaseIdentifiableObject;
-import org.hisp.dhis.common.DimensionItemType;
-import org.hisp.dhis.common.DxfNamespaces;
-import org.hisp.dhis.common.InterpretableObject;
-import org.hisp.dhis.common.MetadataObject;
-import org.hisp.dhis.common.VersionedObject;
+import org.hisp.dhis.common.*;
```
Comment (mortenoh) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataset/DataSet.java]: We don't normally do collapsed import statements.. maybe you can change your intellij settings?

**Pull request merged**